src/TuringTutorials.jl

```diff
@@ -1,186 +1,134 @@
 module TuringTutorials
 
-import IOCapture
+import Requires
+import Weave
 
-using IJulia
-using InteractiveUtils
-using Pkg
-using Plots
-using Requires
-using Weave
+import InteractiveUtils
+import Markdown
+import Pkg
 
-const REPO_DIR = string(pkgdir(TuringTutorials))::String
+const REPO_DIR = dirname(@__DIR__)
+const CSS_FILE = joinpath(REPO_DIR, "templates", "skeleton_css.css")
+const LATEX_FILE = joinpath(REPO_DIR, "templates", "julia_tex.tpl")
 
-include("cache.jl")
-include("build.jl")
-
-export build_folder, tutorial_path, folder2filename
-export build, build_and_exit, verify_logs, tutorials, changed_tutorials
-
-# Not building PDF, because it is fragile. Maybe later.
-default_build_list = (:script, :html, :github, :notebook)
-
-cssfile = joinpath(@__DIR__, "..", "templates", "skeleton_css.css")
-latexfile = joinpath(@__DIR__, "..", "templates", "julia_tex.tpl")
+const DEFAULT_BUILD = (:script, :html, :github)
 
 function __init__()
-    @require Plots = "91a5bcdd-55d7-5caf-9e0b-520d859cae80" include("weaveplots.jl")
-end
-
-"""
-    tutorial_path(folder)
-
-Return the absolute path to a tutorial `folder`.
-"""
-tutorial_path(folder) = joinpath(REPO_DIR, "tutorials", folder)
-
-function polish_latex(path::String)
-    # TODO: Is it maybe better to overload https://github.com/JunoLab/Weave.jl/blob/b5ba227e757520f389a6d6e0f2cacb731eab8b12/src/WeaveMarkdown/markdown.jl#L10-L17
-    # and replace the `tex.formula` there? Only negative part is that this of course
-    # will affect all of the markdown parsers, which is not necessarily desirable.
-    txt = open(f -> read(f, String), path)
-    open(path, "w+") do f
-        txt = replace(txt, "\\\\\n" => "\\\\\\\\\n")
-        write(f, txt)
-    end
+    Requires.@require Plots = "91a5bcdd-55d7-5caf-9e0b-520d859cae80" include("weaveplots.jl")
 end
 
-function weave_file(
-    folder, file, build_list=default_build_list;
-    kwargs...
-)
-    tmp = joinpath(tutorial_path(folder), file)
-    Pkg.activate(dirname(tmp))
+function weave(folder::AbstractString, file::AbstractString; out_path_root::AbstractString=pwd(), build::Tuple{Vararg{Symbol}}=DEFAULT_BUILD)
+  if !issubset(build, (:script, :html, :pdf, :github, :notebook)) 
+    throw(ArgumentError("only build types :script, :html, :pdf, :github, and :notebook are supported"))
+  end
+
+  target = joinpath(REPO_DIR, "tutorials", folder, file)
+  @info("Weaving $(target)")
+  
+  # Activate project
+  # TODO: use separate Julia process?
```

Comment:
Really? Is this not managed by the buildkite stuff? If it ever gives problems please consider switching to PlutoStaticHTML. There, separate Julia processes and all that kind of stuff is managed by Pluto.

Reply:
I can speak to the buildkite stuff: the way this repository is set up, each tutorial `.jmd` file that gets modified will be a separate build job in buildkite; [you can see a list of build jobs](https://buildkite.com/julialang/turingtutorials/builds/17) like `tutorials/05-linear-regression` and `tutorials/04-hidden-markov-models`, etc...  Each one of those jobs is a separate Julia process (indeed, they may even be running on separate machines) so I'm not 100% certain what the intent behind this comment is.

Reply:
Yes, buildkite is fine and not the reason for the comment. The motivation is users that weave a single or multiple tutorials. Since it only affects pdf, html, and markdown output I don't expect it to be a major issue though.